pkg/cache/internal/deleg_map.go

```diff
@@ -39,19 +40,27 @@ type InformersMap struct {
 	Scheme *runtime.Scheme
 }
 
+// resyncPeriod returns a function which generates a duration each time it is
+// invoked; this is so that multiple controllers don't get into lock-step and all
+// hammer the apiserver with list requests simultaneously.
+func resyncPeriod(resync time.Duration) func() time.Duration {
+	return func() time.Duration {
+		factor := rand.Float64() + 1
```

Comment:
this seems like a *really* big variance.

IMO, it probably makes more sense to say `Resync +/- some percentage`, like `Resync +/- 10%`.